### PR TITLE
mkube admins to control access to the backend

### PR DESCRIPTION
**Pull request by dvaldivia**
This resource will be the base to control access to the mkube backend.

An `admin` will be the one that can issue `./m3` commands, he will need to authenticate himself signing his commands with his secret, initially we can use HMAC for that, and later on move to issue certificates or something stronger.

Upon setup of `mkube` an access/secret key is generated (these keys have nothing to do with MinIO access/secret keys) and stored on kubernetes.

This PR contains the table, and the command to add a new admin.

Subsequent PRs will introduce the authentication mechanism, where for an `m3` command to run, the provided credentials must be present either as an environment variable or in a file.

### CODE REVIEW

File: cmd/m3/admin-add.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			Name:  "email",
+			Value: "",
+			Usage: "user email",
+		},
+	},
+}
+
+// adminAdd is a command to add a cluster admin.
+// sample usage:
+//     m3 admin add "User Name" user@acme.com
```
Comment (cesnietor) [inline cmd/m3/admin-add.go:47]: can you please add this new feature to the readme?

File: cluster/migrations/000001_m3-initial.up.sql [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+create table provisioning.admins
+(
+    id uuid
+        constraint admins_pk
+            primary key,
+    name varchar(256),
+    email varchar(256) not null,
+    access_key         varchar(256)                           not null,
+    sys_created_by varchar(256) not null,
+    sys_date_created timestamptz default now() not null
```
Comment (cesnietor) [inline cluster/migrations/000001_m3-initial.up.sql]: all other tables have `sys_create_date`
```suggestion
    sys_created_date timestamptz default now() not null
```

File: cmd/m3/setup.go [hunk 3]
```diff
@@ -32,6 +34,24 @@ var setupCmd = cli.Command{
...
+		fmt.Println("An admin name is needed")
+		return errMissingArguments
+	}
+
+	if email == "" {
+		fmt.Println("An admin email is needed")
+		return errMissingArguments
+	}
+	cluster.SetupM3(name, email)
 	return nil
```
Comment (cesnietor) [inline cmd/m3/setup.go]:
```suggestion
	return cluster.SetupM3(name, email)
```

**Review by cesnietor — CHANGES_REQUESTED** [review]

File: cmd/m3/admin-add.go (hunk 1 above)
Comment (dvaldivia) [reply cmd/m3/admin-add.go:47]: done

File: cmd/m3/setup.go [hunk 4]
```diff
@@ -32,6 +34,24 @@ var setupCmd = cli.Command{
...
+	if name == "" {
+		fmt.Println("An admin name is needed")
+		return errMissingArguments
+	}
+
+	if email == "" {
+		fmt.Println("An admin email is needed")
+		return errMissingArguments
+	}
+	cluster.SetupM3(name, email)
```
Comment (cesnietor) [inline cmd/m3/setup.go]: oh lol it's being called twice
```suggestion
```

Comment (dvaldivia) [reply cmd/m3/setup.go]: lol, I saw your suggestion and I only thought "yeah retuning the error makes sense" hahaha

**Review by cesnietor — APPROVED** [review]

**Pull request merged**